src/lib.rs

```diff
@@ -263,8 +272,16 @@ pub extern "C" fn recvfrom(
     addr: *mut sockaddr,
     addr_len: *mut socklen_t,
 ) -> isize {
-}
+    if addr.is_null() && addr_len.is_null() {
+        return recv(fd, buf, n, flags);
+    }
 
+    return match INDEX.get(fd) {
+        None => next!(recvfrom(fd, buf, n, flags, addr, addr_len)),
+        Some(s) => -1,
+    };
+}
```

Comment:
This code does not match the behavior of the original function: https://github.com/enarx/tlssock/blob/master/lib/tlssock.c#L234